### PR TITLE
Support microprofile-config.properties on JDT LS side

### PR DESCRIPTION
**Pull request by angelozerr**
Support microprofile-config.properties on JDT LS side

See https://github.com/redhat-developer/vscode-quarkus/issues/181

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
This PR gives the capability to load properties from `META-INF/microprofile-config.properties` at first and `application.properties` file after.

To test it, you can play with the `quarkus.http.port` property and codelens URL. 

 * you set this property to 8081 in  `src/main/resources/META-INF/microprofile-config.properties` you refresh the Java editor and you should see CodeLens url with 8081.

 * you set this property to 8082 in  `src/main/resources/application.properties` you refresh the Java editor and you should see CodeLens url with 8082.

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/project/IConfigFile.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+*******************************************************************************/
+package com.redhat.microprofile.jdt.internal.core.project;
+
+/**
+ * Configuration file API
+ * 
+ * @author Angelo ZERR
+ *
+ */
+public interface IConfigFile {
```
Comment (fbricon) [inline microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/project/IConfigFile.java]: IConfigSource

Comment (angelozerr) [reply microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/project/IConfigFile.java]: @fbricon updated

**Comment by angelozerr** [comment]
@fbricon test are fixed now.

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/PropertiesManagerForJava.java [hunk 2]
```diff
@@ -268,10 +138,10 @@ private static ITypeRoot resolveTypeRoot(String uri, IJDTUtils utils, IProgressM
 		return unit != null ? unit : classFile;
 	}
 
-	private JDTQuarkusProjectInfo getJDTQuarkusProjectInfo(IJavaProject project) throws JavaModelException {
-		JDTQuarkusProjectInfo info = infos.get(project);
+	private JDTMicroProfileProject getJDTQuarkusProjectInfo(IJavaProject project) throws JavaModelException {
```
Comment (fbricon) [inline microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/PropertiesManagerForJava.java]: getJDTMicroProfileProject if you want to stay consistent

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/project/AbstractConfigSource.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+	 * capability to get the filtered value.
+	 * </p>
+	 * 
+	 * @return the target/classes/$configFile and null otherwise.
+	 */
+	private Path getConfigFile() {
+		if (configFile != null && Files.exists(configFile)) {
+			return configFile;
+		}
+		try {
```
Comment (fbricon) [inline microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/project/AbstractConfigSource.java]: make sure IJavaProject (getProject) is still accessible

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/project/JDTMicroProfileProject.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+/**
+ * JDT MicroProfile project.
+ * 
+ * @author Angelo ZERR
+ *
+ */
+public class JDTMicroProfileProject {
+
+	public static final String MICROPROFILE_CONFIG_PROPERTIES_FILE = "META-INF/microprofile-config.properties";
+	public static final String APPLICATION_PROPERTIES_FILE = "application.properties";
```
Comment (fbricon) [inline microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/project/JDTMicroProfileProject.java]: those will need to be decoupled eventually

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/project/JDTMicroProfileProject.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+	private final List<IConfigSource> configSources;
+
+	private static final String QUARKUS_HTTP_PORT = "quarkus.http.port";
```
Comment (fbricon) [inline microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/project/JDTMicroProfileProject.java]: will need to be decoupled eventually, as it's a quarkus thing, not an MP thing

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/PropertiesManagerForJava.java (hunk 2 above)
Comment (angelozerr) [reply microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/PropertiesManagerForJava.java]: Fixed and move this method in a new class JDTMicroProfileProjectManager (more clean I find)

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/project/JDTMicroProfileProject.java (hunk 4 above)
Comment (angelozerr) [reply microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/project/JDTMicroProfileProject.java]: what do you mean with decoupled ?

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/project/AbstractConfigSource.java (hunk 3 above)
Comment (angelozerr) [reply microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/project/AbstractConfigSource.java]: @fbricon updated

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/project/JDTMicroProfileProject.java (hunk 5 above)
Comment (angelozerr) [reply microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/project/JDTMicroProfileProject.java]: I agree we should manage that with new extension point (like we have done for properties provider), but I think we should create a new issue for that which is not trivial.

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/project/JDTMicroProfileProject.java (hunk 4 above)
Comment (angelozerr) [reply microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/project/JDTMicroProfileProject.java]: I agree we should manage that with new extension point (like we have done for properties provider), but I think we should create a new issue for that which is not trivial.

**Pull request merged**